### PR TITLE
194 region stencils (Supersedes PR #1046)

### PR DESCRIPTION
**Pull request by andrewcoughtrie**
Created a branch to deal with the accidental inclusion of submodule data. Supersedes PR #1046. 

Responding to the [review](https://github.com/stfc/PSyclone/pull/1046#pullrequestreview-570479752) on #1046.

### CODE REVIEW

**Comment by andrewcoughtrie** [comment]
@arporter I've put the changes requested in PR#1046 in their own commits to hopefully make it easier for you to review them.

**Comment by andrewcoughtrie** [comment]
Had to fix pycodestyle issues I forgot about, done now.

**Review by arporter — CHANGES_REQUESTED** [review]
Thanks for making those changes, especially for moving the stencil tests - I hadn't realised just how big a job that was.
Updated documentation builds fine and looks good.
Tests and examples all OK with compilation.
There are just a few pylint things to fix in the new test file and then this can be merged.

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py [hunk 1]
```diff
@@ -0,0 +1,1832 @@
...
+''' Module containing tests of LFRic stencils through the Dynamo0.3 API '''
+
+# imports
+from __future__ import absolute_import, print_function
+
+import os
+
+import fparser
+from fparser import api as fpapi
+import pytest
```
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:45]: I don't get any complaints from pylint about this, I can reorder them but I don't know what order it wants. Is it complaining about the order of the imports overall?

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py [hunk 2]
```diff
@@ -0,0 +1,1832 @@
...
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. Coughtrie, Met Office
```
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:34]: Done.

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py [hunk 3]
```diff
@@ -0,0 +1,1832 @@
...
+
+''' Module containing tests of LFRic stencils through the Dynamo0.3 API '''
```
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:36]: Not sure what you mean for this. "Module containing tests of LFRic stencils through the **s/Dynamo0.3/LFRic/** API" is that what I should change it to?

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py [hunk 4]
```diff
@@ -0,0 +1,1832 @@
...
+from psyclone.tests.lfric_build import LFRicBuild
+
+# constants
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "../..", "test_files", "dynamo0p3")
+# Get the root directory of this PSyclone distribution
+ROOT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(
+    os.path.dirname(os.path.abspath(__file__)))))
+# Construct the path to the default configuration file
+DEFAULT_CFG_FILE = os.path.join(ROOT_PATH, "config", "psyclone.cfg")
```
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:63]: Deleted.

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py [hunk 5]
```diff
@@ -0,0 +1,1832 @@
...
+from psyclone.parse.algorithm import parse
+from psyclone.parse.utils import ParseError
+from psyclone.psyGen import PSyFactory
+from psyclone.tests.lfric_build import LFRicBuild
+
+# constants
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "../..", "test_files", "dynamo0p3")
+# Get the root directory of this PSyclone distribution
+ROOT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(
```
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:60]: Deleted

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:36]: @TeranIvy explained your use of sed short hand, I get it now.

[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:36]: Apologies - I forgot I might not have used that on you before!

File: src/psyclone/tests/domain/lfric/lfric_stencil_tests.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:45]: `domain/lfric/lfric_stencil_tests.py:45:0: C0411: third party import "import pytest" should be placed before "import fparser" (wrong-import-order)`

[the remark this answers is not in the document]
Comment (andrewcoughtrie) [reply src/psyclone/tests/domain/lfric/lfric_stencil_tests.py:45]: Done

**Review by arporter — APPROVED** [review]
All requested changes have been made. Proceeding to merge.

**Pull request merged**